transaction-context/src/lib.rs

```diff
@@ -827,8 +829,22 @@ impl InstructionContext {
         self.try_borrow_account(
             transaction_context,
             index_in_transaction,
-            self.get_number_of_program_accounts()
-                .saturating_add(instruction_account_index),
+            Some(instruction_account_index),
+        )
+    }
+
+    /// Retrieves an instruction account using the index in transaction
+    pub fn try_borrow_instruction_account_with_transaction_index<'a, 'b: 'a>(
```

Comment:
I would say we inline this at its three callsites. That way it is clear where the expensive `get_index_of_account_in_instruction()` happens.

Reply:
Fixed in https://github.com/anza-xyz/agave/pull/7220/commits/16d184283ce35aa78d7869bdc5a21132e709d86a